plugins/filter/from_csv.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Andrew Pantuso (@ajpantuso) <ajpantuso@gmail.com>
+# Copyright: (c) 2018, Dag Wieers (@dagwieers) <dag@wieers.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import csv
+from io import BytesIO, StringIO
+
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import PY3
+
+
+# Add Unix dialect from Python 3
```

Comment:
It might make sense to move the common parts into a `csv` module_utils. That way, the module and the filter plugin can share common code, instead of having duplicate copies of essentially the same code.

Reply:
It absolutely makes sense to do that I just didn't want to refactor the read_csv module just to introduce this filter.
Would it make sense for me to pull out the interfaces I think are reusable (Dialect Initializer, CSV reader, Error handler, etc...) and only introduce them into this filter with this PR?
Then adapting read_csv to use the common bits would be another PR?

Reply:
@Ajpantuso that would be fine for me. Also updating csv in the same PR would also be fine for me. Whatever you prefer :)

One advantage of involving the csv module is that everybody interested in the csv module will know about it, and it shouldn't happen that someone modified the csv module before your PR got merged and then you might miss that change when updating the csv module next.